### ISSUE TITLE
Getting RollupJob error: "ERROR at Row:1:Column:49 Invalid field: 'Name'. Review Rollup Summary Schedule Items"

### ISSUE THREAD

**Issue by lrock-eqt** [SFDO-Community/declarative-lookup-rollup-summaries#1485]
**Describe the bug**
I'm not sure what's going on, but running the RollupJob is resulting in an invalid field error on the 'Name' field. This is a field for which permissions are automatically granted when you have access to an object, so I need some help understanding what could possibly going wrong.

This is replicatable in production and full sandbox.

**To Reproduce**

1. Create a Lookup Rollup Summary on Account that uses a Calculation Mode of "Realtime" and activate it (may or may not need both the Parent and Child objects to be "Account").
2. Merge two accounts to generate a "Lookup Rollup Summary Schedule Item".
3. Run the RollupJob.
4. Observe an error email is received:
Error:
dlrs__LookupRollupSummary2__c, dlrs__ParentId__c, dlrs__ParentKeyPrefix__c
                                   ^
ERROR at Row:1:Column:49
Invalid field: 'Name'. Review Rollup Summary Schedule Items 

**Expected behavior**
No error should be encountered on the Name field as this field is automatically granted when the user has access to the "Lookup Rollup Summary Schedule Item" object and the fields are being dynamically queried based on user access.

**Screenshots**
![LookupRollupSummaryScheduleItem_InvalidFieldName](https://github.com/SFDO-Community/declarative-lookup-rollup-summaries/assets/103588445/857a58c1-83b0-4f3d-8f9a-25dd09eda05f)

![erroremail](https://github.com/SFDO-Community/declarative-lookup-rollup-summaries/assets/103588445/ba01966b-861d-4466-8ab3-0799bb799a66)

**Additional context**
N/A

**Comment by aheber** [comment]
@lrock-eqt we would recommend that you first post in the [Trailblazer Community Group](https://success.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F9300000009O5p) when you run into problems. There are a great many more people that watch that group who could be of assistance to you.

As for this, I haven't seen this error before in this context. I would ensure your user has the `Lookup Rollup Summaries - Configure Rollups` permission set assigned as a first place to look.

Aside from that I would use Debug Logging to try and catch this in the act, from there you should be able to pull out the SOQL_EXECUTE log line that represents the actual attempted query, we can look at see what it is doing and maybe diagnose from there.

Additionally, I see that you've sent a rollup configuration screenshot. Does that represent the only configuration that is present in your scheduled items or are there others at play?

**Comment by lrock-eqt** [comment]
@aheber I ran a lot of tests and I believe I've identified the source of the issue. I found a set of 26 records that were causing a problem.

I ran the records in groups where they had the same value for the "Name" field and I do not get the error. 
I ran the records in groups where they had the same value for the "dlrs__LookupRollupSummary2__c" field and did not get the error.

I ran the records in groups where they had the same value for the "dlrs__ParentId__c" field and I _did_ get the error.

This is repeatable with multiple combinations where Name and dlrs__LookupRollupSummary2__c are unique but dlrs__ParentId__c is identical. 
Here are two sample records that caused the error when existing in the set at the same time:
[LookupRollupSummaryScheduleItems_SampleErrorLines_SameParent.csv](https://github.com/user-attachments/files/16353648/LookupRollupSummaryScheduleItems_SampleErrorLines_SameParent.csv)

The error is definitely happening in the RollupJob try/catch from lines 45 - 79, specifically in the RollupService.processScheduleItems method. The job fails for **ALL** "Lookup Rollup Summary Schedule Item" records in scope, even those on other objects that do not display the conflict noted above, which to my understanding means that the issue has to be manually resolved every time it is encountered in any two records in the set. Evaluating these records without this "All or None" concept would make the issue much easier to resolve and would cause much fewer headaches.

I should note that I have only replicated this error for Accounts that were merged, where the DLRS's have a calculation mode of "Realtime", so for us it is a unique scenario but definitely one that we will encounter often, as we do a lot of duplicate management on Accounts.
